### PR TITLE
CP-23986: Implement QMP event thread in xenopsd

### PR DESCRIPTION
**Pull request by fillzero**
Signed-off-by: Liang Dai <liang.dai1@citrix.com>

### CODE REVIEW

**Comment by fillzero** [comment]
@mg12 @lindig  Would you kindly to have a look?

I use `select` here, I know use `epoll` should be better, but the ocaml library doesn't support it. 
Or should I write a `c` sub program includes `epoll"?

File: xc/device.ml [hunk 1]
```diff
@@ -1950,3 +1950,66 @@ let get_tc_port ~xs domid =
 	if qemu_exists
 	then Dm.get_tc_port ~xs domid
 	else PV_Vnc.get_tc_port ~xs domid
+
+let qmp_event_tbl = Hashtbl.create 1000
```
Comment (lindig) [inline xc/device.ml]: What is the difference between `qmp_event` and `event_qmp` and what is the purpose of these two tables? Is there a design doc?

**Comment by jonludlam** [comment]
Hi @fillzero - this looks very much like a work-in-progress. It's probably better to get feedback at this sort of level elsewhere - your own github fork, for example, and then submit the pull request when it's ready for inclusion. Additionally, do we have a design doc for this anywhere?

**Comment by fillzero** [comment]
@jonludlam Thanks for your comments, I will follow your advice. 
This is `upstream qemu` feature, and the design doc doesn't cover much about QMP with toolstack.
But we need QMP events to trigger some callback functions.

**Pull request closed without merging**